tests/unit/test_p11wrapper.cpp

```diff
@@ -0,0 +1,400 @@
+/*
```

Comment:
Only gave the tests a quick glance…

Reply:
That's fair. Will be breaking this PR down to smaller multiple one anyways...